server/application/application.go

```diff
@@ -1118,6 +1135,12 @@ func (s *Server) PodLogs(q *application.ApplicationPodLogsQuery, ws application.
 						close(done)
 						return
 					}
+					if q.Filter != nil {
```

Comment:
The filter should be applied before checking time. Overwise user might get an empty response while paging back with applied filter